### PR TITLE
Add support for Nokia ISAM

### PR DESCRIPTION
**Pull request by vitalisator**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by murrant** [comment]
Can you remove the commented lines of code?
A multi-get on for the device os polling is more efficient (but some devices don't like that).

**Comment by vitalisator** [comment]
@murrant, done

**Comment by murrant** [comment]
@vitalisator did you intentionally remove the copyright/license comments?

**Comment by vitalisator** [comment]
> @vitalisator did you intentionally remove the copyright/license comments?
no, I was not sure which comments did you meant in your last post :-)

**Comment by vitalisator** [comment]
@murrant , thanks. I have overseen that. But Travis is still failed?

**Comment by PipoCanaja** [comment]
Some test data must be cleaned as shown by travis. Tests were identifying devices as timos and should now change to nokia-isam.
You can run scripts/pre-commit.php -o timos and scripts/pre-commit.php -o nokia-isam and correct anything that must be corrected.

**Comment by murrant** [comment]
I think the discovery is still wrong.  Right now it says all Nokia devices that aren't `1.3.6.1.4.1.6527.1` are ISAM.

It seems like Nokia is pretty good about the sysDescr, I think it should be just:
```yaml
    - sysDescr: NOKIA ISAM
    - sysObjectID:
        - .1.3.6.1.4.1.637.61
```

File: includes/discovery/sensors/pre-cache/nokia-isam.inc.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            unset($twopart_value);
+        }
+    }
+}
+
+// temperature pre cache
+$pre_cache['nokiaIsamSlotTemperature'] = snmpwalk_cache_twopart_oid($device, 'eqptBoardThermalSensorTable', [], 'ASAM-EQUIP-MIB', 'nokia');
+
+// state pre cache
+$pre_cache['nokiaIsamEqpBoardTable'] = snmpwalk_cache_oid($device, 'eqptBoardTable', [], 'ASAM-EQUIP-MIB', 'nokia', '-OQUse');
```
Comment (murrant) [inline includes/discovery/sensors/pre-cache/nokia-isam.inc.php]: I don't like use of pre-cache this way.  You can't disable pre-cache, it is run for every poll (even if the sensors module is disabled).  So, if only one module uses snmp data, the fetch should be in that module.

**Comment by vitalisator** [comment]
not existing Test for bad_iftype in yaml file?
nokia-isam.yaml does not validate. Violations:
[] The property bad_iftype is not defined and the definition does not allow additional properties

**Comment by murrant** [comment]
@vitalisator the schema is incomplete built to accommodate existing definitions (then expanded some). I'd appreciate it if you updated the schema.

**Comment by murrant** [comment]
I updated the schema

**Comment by murrant** [comment]
Updated to use ISO 8601 format

**Pull request merged**